core/debugger/remote_debugger.cpp

```diff
@@ -141,7 +141,7 @@ void RemoteDebugger::_print_handler(void *p_this, const String &p_string, bool p
 	}
 
 	if (allowed_chars < s.length()) {
-		s = s.substr(0, allowed_chars);
+		s = s.left(allowed_chars);
```

Comment:
First, allowed_chars is defined as:

`int allowed_chars = MIN(MAX(rd->max_chars_per_second - rd->char_count, 0), s.length());`

So one of the minimums is a non-negative length, and the other is the MAX of something and 0, which is at least 0. Since `allowed_chars` is not modified after, it is always positive at this point.